File: src/main/java/com/aquasecurity/plugins/trivy/ui/treenodes/VulnerablePackageTreeNode.java
package com.aquasecurity.plugins.trivy.ui.treenodes;

import com.aquasecurity.plugins.trivy.model.Vulnerability;

import javax.swing.*;
import javax.swing.tree.DefaultMutableTreeNode;

class VulnerablePackageTreeNode extends DefaultMutableTreeNode implements TrivyTreeNode {

    public final Vulnerability vulnerability;

    public VulnerablePackageTreeNode(Vulnerability vulnerability) {
        super(vulnerability);
        this.vulnerability = vulnerability;
    }

    @Override
    public Icon getIcon() {
        return null;
    }

    @Override
    public String getTitle() {
        return this.vulnerability.pkgName;
    }

    @Override
    public String getTooltip() {
        return this.vulnerability.description;
    }
}

File: src/main/kotlin/com/aquasecurity/plugins/trivy/ui/treenodes/VulnerablePackageTreeNode.kt
package com.aquasecurity.plugins.trivy.ui.treenodes

import com.aquasecurity.plugins.trivy.model.Vulnerability
import javax.swing.Icon
import javax.swing.tree.DefaultMutableTreeNode

internal class VulnerablePackageTreeNode(val vulnerability: Vulnerability) :
    DefaultMutableTreeNode(vulnerability), TrivyTreeNode {
    override val icon: Icon?
        get() = null

    override val title: String
        get() = vulnerability.pkgName!!

    override val tooltip: String
        get() = vulnerability.description!!
}